pkgs/development/haskell-modules/lib.nix

```diff
@@ -76,6 +76,14 @@ rec {
     fixupPhase = ":";
   });
 
+  # link executables statically against haskell libs to reduce closure size
+  justStaticExecutables = drv: overrideCabal drv (drv: {
+    enableSharedExecutables = false;
+    isLibrary = false;
+    doHaddock = false;
+    postFixup = "rm -rf $out/lib $out/nix-support $out/share/doc";
```

Comment:
Should we really delete `$out/share/doc` here? What if the package includes manpages or other user-facing documentation? We can probably delete `$out/share/doc/*-ghc-${ghc.version}` though, but since haddocks are disabled, shouldn't that not exist already?